agent/api/task/task.go

```diff
@@ -1882,6 +1987,46 @@ func (task *Task) shouldOverrideNetworkModeAwsvpc(container *apicontainer.Contai
 	return true, dockerMappingContainerPrefix + pauseContainer.DockerID
 }
 
+// shouldOverrideNetworkModeServiceConnectBridge checks if a bridge-mode SC task container needs network mode override
+// For non-internal containers in an SC bridge-mode task, each gets a pause container, and should be launched
+// with container network mode use pause container netns (the "docker run" equivalent option is
+// "--network container:$pause_container_id")
+func (task *Task) shouldOverrideNetworkModeServiceConnectBridge(container *apicontainer.Container, dockerContainerMap map[string]*apicontainer.DockerContainer) (bool, string) {
+	if container == task.GetServiceConnectContainer() {
+		return false, "" // SC container will use bridge mode (it doesn't need a pause container)
+	}
+	pauseContName, ok := task.ServiceConnectPauseContainerMapping[container.Name]
+	if !ok || pauseContName == "" {
+		// This should never be the case and implies a code-bug.
+		logger.Critical("Pause container required per task container for Service Connect task bridge mode, but " +
```

Comment:
can we double check that `Critical` doesn't also terminate the program?

Reply:
We use seelog which will return an error for `Error`, `Warn`, and `Critical`, but no it does not terminate the program

https://github.com/aws/amazon-ecs-agent/blob/master/agent/vendor/github.com/cihub/seelog/log.go